Libraries/CustomComponents/Navigator/Navigator.js

```diff
@@ -1080,8 +1080,7 @@ var Navigator = React.createClass({
     var newRenderedSceneMap = new Map();
     var scenes = this.state.routeStack.map((route, index) => {
       var renderedScene;
-      if (this._renderedSceneMap.has(route) &&
-          index !== this.state.presentedIndex) {
+      if (this._renderedSceneMap.has(route)) {
         renderedScene = this._renderedSceneMap.get(route);
```

Comment:
So you'd get the same old copy of rendered scene even when `this.state.presentedIndex` had changed.
Some people may rely on using the the index to determined whether a scene is active or not and this change will break that.

If the scene can be expensive to render, the scene itself should use PureRenderMixin to protect itself from being overly rendered by the navigator.